### PR TITLE
CSPL-1727: Manifest files to differentiate namespace scoped and cluster scoped

### PR DESCRIPTION
**Pull request by vivekr-splunk**
Now there will 2 different manifest files generated for release. (as earlier)
1. namespace scoped - `splunk-operator-install.yaml`
2. cluster scoped - `splunk-operator-cluster.yaml`

### CODE REVIEW

**Review by mgaldino-splunk — APPROVED** [review]

File: config/default/kustomization-cluster.yaml [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+commonLabels:
+  name: splunk-operator
+
+bases:
+- ../crd
+- ../rbac
+#- ../persistent-volume
+- ../service
+- ../manager
+# [WEBHOOK] To enable webhook, uncomment all the sections with [WEBHOOK] prefix including the one in
```
Comment (akondur) [inline config/default/kustomization-cluster.yaml:21]: Do we need all these commented lines for later?

File: config/default/kustomization-namespace.yaml [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+commonLabels:
+  name: splunk-operator
+
+bases:
+- ../crd
+- ../rbac
+#- ../persistent-volume
+- ../service
+- ../manager
+# [WEBHOOK] To enable webhook, uncomment all the sections with [WEBHOOK] prefix including the one in
```
Comment (akondur) [inline config/default/kustomization-namespace.yaml:21]: Same as above?

**Review by jryb — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -133,8 +136,14 @@ uninstall: manifests kustomize ## Uninstall CRDs from the K8s cluster specified
 	$(KUSTOMIZE) build config/crd | kubectl delete -f -
 
 deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
+	sed -i "" "s/namespace: splunk-operator/namespace: ${NAMESPACE}/g"  config/default/kustomization.yaml
+	sed -i "" "s/value: WATCH_NAMESPACE_VALUE/value: \"${WATCH_NAMESPACE}\"/g"  config/default/kustomization.yaml
+	sed -i "" "s|SPLUNK_ENTERPRISE_IMAGE|${SPLUNK_ENTERPRISE_IMAGE}|g"  config/default/kustomization.yaml
```
Comment (sgontla) [inline Makefile:141]: Just for my understanding, what is this particular change for?

File: Makefile [hunk 4]
```diff
@@ -133,8 +136,14 @@ uninstall: manifests kustomize ## Uninstall CRDs from the K8s cluster specified
 	$(KUSTOMIZE) build config/crd | kubectl delete -f -
 
 deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
+	sed -i "" "s/namespace: splunk-operator/namespace: ${NAMESPACE}/g"  config/default/kustomization.yaml
+	sed -i "" "s/value: WATCH_NAMESPACE_VALUE/value: \"${WATCH_NAMESPACE}\"/g"  config/default/kustomization.yaml
```
Comment (sgontla) [inline Makefile:140]: Does this sed format work on Linux? I think it may fail, as the -i options are different. Same comment for other places below.

Comment (vivekr-splunk) [reply Makefile:140]: I thought I fixed it . sorry its for mac, I wanted to add for mac and linux , let me fix it

File: Makefile (hunk 3 above)
Comment (vivekr-splunk) [reply Makefile:141]: Idea is to take splunk enterprise image version on command line as argument and then pass it while generating the yaml file. 
`make generate-artifacts IMG=splunk/splunk-operator:1.1.0 SPLUNK_ENTERPRISE_IMAGE=splunk/splunk:8.2.6 WATCH_NAMESPACE="" VERSION=1.1.0 NAMESPACE=splunk-operator`

File: config/default/kustomization-cluster.yaml (hunk 1 above)
Comment (vivekr-splunk) [reply config/default/kustomization-cluster.yaml:21]: commented lines are for us to use when we start using webhooks, and patches . these are auto generated code, I am not removing them. if someone is using code to build they can use the webhooks or patches if needed. the code helps in customizing the operator.

**Review by akondur — APPROVED** [review]

File: Makefile [hunk 5]
```diff
@@ -68,6 +71,15 @@ else
...
 endif
 
+SED := sed -i 
+ifeq ($(shell uname), Linux)
+	SED = sed -i  
+else ifeq ($(shell uname), Darwin)
+	SED = sed -i "" 
+else
+	SED = sed -i  
+endif
```
Comment (sgontla) [inline Makefile:81]: Thanks for addressing this, this will avoid confusion.

Comment (sgontla) [inline Makefile:81]: nit: you are already starting with default value.

**Review by sgontla — APPROVED** [review]

**Pull request merged**